Source/Core/DiscIO/VolumeCommon.cpp

```diff
@@ -165,4 +165,409 @@ u8 GetSysMenuRegion(u16 _TitleVersion)
 	}
 }
 
+#define COMPANY_ID(first_char, second_char) ((u16)first_char << 8) + second_char
+std::string GetCompanyFromID(const std::string& company_id)
+{
+	static const std::map<u16, std::string> companies =
```

Comment:
Prior to a rebase which steamrolled comments, @heinermann suggested just doing a string → string key-value setup. I think it would be fine to do it that way. There's really no need to micro-optimize (especially without benchmarks that indicate the straightforward way is dramatically slow).